src/Data/Vec/Properties/Heterogeneous.agda

```diff
@@ -0,0 +1,210 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some Vec-related properties
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Vec.Properties.Heterogeneous where
+
+open import Data.Nat.Base
+open import Data.Nat.Properties using (+-assoc; +-suc)
+open import Data.Product as Prod
+  using (_×_; _,_; Σ; Σ-syntax; proj₁; proj₂)
+open import Data.Vec.Base
+open import Data.Vec.Properties
+-- open import Function.Base
+-- open import Function.Inverse using (_↔_; inverse)
+open import Level using (Level)
+open import Relation.Binary.PropositionalEquality as P
+  using (_≡_; _≢_; refl; _≗_; cong₂)
+open P.≡-Reasoning
+
+private
+  variable
+    a b c d : Level
+    A : Set a
+    B : Set b
+    C : Set c
+    D : Set d
+
+------------------------------------------------------------------------
+-- Heterogeneous propositional equality over vectors
+
+data ≡ᵥ {a} {A : Set a} : {m : ℕ} (xs : Vec A m) {n : ℕ} (ys : Vec A n) → Set (Level.suc a) where
```

Comment:
So there's a lot of great work in this file, but I think that the crux of this definition already exists in `Data.Vec.Relation.Binary.Pointwise.Inductive`. If you then look at it's instantiation in `Data.Vec.Relation.Binary.Equality.Propositional` to a propositional setoid then, modulo requiring a proof that the heads of the lists are equal, the resulting definition of `_≋_` is identical to this one.

Any chance you could you port the contents to the correct places in those two files?

Reply:
OK, I suspect that's going to require a bit more time and patience as I work my way round those other definitions. I'll focus on the main (homogeneous) properties first, so you'll want to hang on a bit for this collection of commits ;-)

Reply:
Hmmm. I realise I may have once again done things out-of-order...
... because you won't be able to close the issue with this PR unless these problems with `Heterogeneous` are fixed first. 
Should I push a commit which removes that file (and you could then merge the 'reduced' PR), and then add it back with a view to resolving this (non-trivial?) issue of reorganisation? Hmmm. Hmmm. Hmmm. 
Food needed!

Reply:
Yes, that sounds like a good plan :+1:

Reply:
OK, will do that now.

Reply:
So... is it the case that the relation defined in `Data.Vec.Relation.Binary.Pointwise.Inductive` is heterogeneous wrt the lengths of the `Vec`s concerned? I perhaps hadn't previously appreciated that (and in any case, while developing new stuff, I still find it easier to start from a fresh context... not perhaps the best way to do library development, though ;-))

Reply:
OK, I think I will need a bit more hand-holding about the library structure here before embarking on the `Heterogeneous` commits... but hopefully you can merge this PR anyway ahead of doing that, while still leaving issue #942 open.

Reply:
Sounds like a plan!